src/ape/api/providers.py

```diff
@@ -926,7 +926,10 @@ def _prepare_call(self, txn: TransactionAPI, **kwargs) -> List:
             if value is not None and not isinstance(value, str):
                 txn_dict[field] = to_hex(value)
 
-        block_identifier = kwargs.pop("block_identifier", "latest")
+        if "block_identifier" in kwargs:
```

Comment:
For improving / adding a test, see this test as an example: https://github.com/ApeWorX/ape/blob/main/tests/functional/test_contract_instance.py#L53-L54